### PR TITLE
Feature 22 Selector Component

### PR DESCRIPTION
**Pull request by VottonDev**
Selector component used by #20

### CODE REVIEW

**Comment by VottonDev** [comment]
Looks currently like this:
image

**Comment by VottonDev** [comment]
This is what it looks like now:
image

**Review by EVP1989 — COMMENTED** [review]
Added few suggestions and comment

File: src/components/selectors/user-selector.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { TextField } from "@mui/material";
+import { Selector, FormControlWrapper } from "../../styled/layouts/admin-layout";
+import strings from "localization/strings";
+
+/**
+ * Disable second selector by default and if first selector is selected, enable second selector
+*/
+const UserSelector: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (VottonDev) [reply src/components/selectors/user-selector.tsx:9]: I'll rename it to PermissionFilter 👍

File: src/components/selectors/permission-filter.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+*/
+const PermissionFilter: React.FC = () => {
+  const [ firstSelector, setFirstSelector ] = React.useState<string>("");
+  const [ secondSelector, setSecondSelector ] = React.useState<string>("");
+
+  /**
+   * We need to disable second selector if first selector is selected
+   * We read that the first selector has "" as value, so we need to check if it is empty
+   * @param event - event object
+   */
```
Comment (EVP1989) [inline src/components/selectors/permission-filter.tsx:17]:
```suggestion
  /**
   * We need to disable second selector if first selector is selected
   * We read that the first selector has "" as value, so we need to check if it is empty
   *
   * @param event - event object
   */
```

Comment (EVP1989) [reply src/components/selectors/permission-filter.tsx:17]: Btw, are you using linter? It should complain about these.

File: src/components/selectors/permission-filter.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  const handleFirstSelectorChange = (event: React.ChangeEvent<{ value: string }>) => {
+    setFirstSelector(event.target.value);
+    setSecondSelector("");
+  };
+
+  /**
+   * We enable second selector if first selector is selected
+   * @param event - event object
+   */
```
Comment (EVP1989) [inline src/components/selectors/permission-filter.tsx:26]:
```suggestion
  /**
   * We enable second selector if first selector is selected
   *
   * @param event - event object
   */
```

File: src/components/selectors/permission-filter.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  const handleSecondSelectorChange = (event: React.ChangeEvent<{ value: string }>) => {
+    setSecondSelector(event.target.value);
+  };
+
+  return (
+    <FormControlWrapper>
+      <Selector
```
Comment (EVP1989) [inline src/components/selectors/permission-filter.tsx:33]: These could also be renamed, maybe FilterSelector is more describing.

**Review by EVP1989 — COMMENTED** [review]
Added suggestions and comment

File: src/components/selectors/permission-filter.tsx (hunk 4 above)
Comment (VottonDev) [reply src/components/selectors/permission-filter.tsx]: Changed the names now, so both have the word Filter in them as well

**Review by EVP1989 — APPROVED** [review]

**Pull request merged**